From 70441611d4e8200d9d16dfed493873b8c1bb57c5 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 15 Mar 2021 11:33:38 -0700
Subject: [PATCH] Remove HAVE_GETCONTEXT check to add local implementation

On musl getcontext/setcontext APIs are implemented in libucontext which
can be used

Upstream-Status: Inappropriate [Musl Specific]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makefile.am | 12 ------------
 1 file changed, 12 deletions(-)

--- a/Makefile.am
+++ b/Makefile.am
@@ -529,10 +529,6 @@ src_client_linux_libbreakpad_client_a_SO
 	src/common/linux/linux_libc_support.cc \
 	src/common/linux/memory_mapped_file.cc \
 	src/common/linux/safe_readlink.cc
-if !HAVE_GETCONTEXT
-src_client_linux_libbreakpad_client_a_SOURCES += \
-	src/common/linux/breakpad_getcontext.S
-endif
 
 # Client tests
 src_client_linux_linux_dumper_unittest_helper_SOURCES = \
@@ -580,10 +576,6 @@ src_client_linux_linux_client_unittest_s
 	src/processor/minidump.cc \
 	src/processor/pathname_stripper.cc \
 	src/processor/proc_maps_linux.cc
-if !HAVE_GETCONTEXT
-src_client_linux_linux_client_unittest_shlib_SOURCES += \
-	src/common/linux/breakpad_getcontext.S
-endif
 
 src_client_linux_linux_client_unittest_shlib_CPPFLAGS = \
 	$(AM_CPPFLAGS) $(TEST_CFLAGS)
@@ -613,10 +605,6 @@ src_client_linux_linux_client_unittest_s
 	src/common/string_conversion.o \
 	$(TEST_LIBS) \
 	$(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
-if !HAVE_GETCONTEXT
-src_client_linux_linux_client_unittest_shlib_SOURCES += \
-	src/common/linux/breakpad_getcontext_unittest.cc
-endif
 if ANDROID_HOST
 src_client_linux_linux_client_unittest_shlib_LDFLAGS += \
 	-llog -lm
